### PR TITLE
cmd/configure.ac: rework

### PR DESCRIPTION
**Pull request by valentindavid**
This is a cleanup of `cmd/configure.ac`.
 * Cache results from running compiler.
 * Use `AX_APPEND_COMPILE_FLAGS` to make sure warning flags exist
 * Wrap test code in `AC_LANG_SOURCE`/`AC_LANG_PROGRAM` and simplify test code.

### CODE REVIEW

File: cmd/configure.ac [hunk 1]
```diff
@@ -22,11 +22,7 @@ AC_SYS_LARGEFILE
 # Checks for header files.
 AC_CHECK_HEADERS([fcntl.h limits.h stdlib.h string.h sys/mount.h unistd.h])
 AC_CHECK_HEADERS([sys/quota.h], [], [AC_MSG_ERROR(sys/quota.h unavailable)])
-AC_CHECK_HEADERS([xfs/xqm.h], [], [AC_MSG_ERROR(xfs/xqm.h unavailable)],
-[[#define _GNU_SOURCE
-#define _FILE_OFFSET_BITS 64
```
Comment (mvo5) [inline cmd/configure.ac:27]: out of curiosity, what macro defines this already?

**Review by mvo5 — APPROVED** [review]
Thank you

File: cmd/configure.ac (hunk 1 above)
Comment (valentindavid) [reply cmd/configure.ac:27]: `AC_LANG_CONFTEST` will prepend `confdefs.h`. All prior `AC_DEFINE*` will add into `confdefs.h`.  `AC_SYS_LARGEFILE` will do and `AC_DEFINE` for `_FILE_OFFSET_BITS`. `AC_USE_SYSTEM_EXTENSIONS` will do one for `_GNU_SOURCE`.

**Review by Meulengracht — APPROVED** [review]
Looks nice, but I would prefer the PR description to have some context as to what we are achieving here

**Pull request merged**